model/llm/llm.go

```diff
@@ -0,0 +1,87 @@
+package llm
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/zimmski/osutil/bytesutil"
+
+	"github.com/symflower/eval-symflower-codegen-testing/model"
+	"github.com/symflower/eval-symflower-codegen-testing/model/prompt"
+	"github.com/symflower/eval-symflower-codegen-testing/provider"
+)
+
+// llm represents a LLM model accessed via a cloud provider.
+type llm struct {
+	// provider is the client to query the LLM model.
+	provider provider.QueryProvider
+	// model holds the identifier string for the LLM model.
+	model string
+}
+
+// NewLLMModel returns an LLM model corresponding to the given identifier which is queried via the given provider.
+func NewLLMModel(provider provider.QueryProvider, modelIdentifier string) model.Model {
+	return &llm{
+		provider: provider,
+		model:    modelIdentifier,
+	}
+}
+
+// llmGenerateTestForFilePrompt is the prompt used to query LLMs for test generation.
+var llmGenerateTestForFilePrompt = bytesutil.StringTrimIndentations(`
+	Given the following Go code file, provide a test file for this code.
+	The tests should produce 100 percent code coverage and must compile.
+	The response must contain only the test code and nothing else.
+`)
+
+// llmGenerateTestForFilePromptContext is the context for for the prompt template.
+type llmGenerateTestForFilePromptContext struct {
+	Prompt string
+	Code   string
+}
+
+// llmGenerateTestForFilePromptData is the template for generating an LLM test generation prompt.
+var llmGenerateTestForFilePromptData = template.Must(template.New("templateGenerateTestPrompt").Parse(bytesutil.StringTrimIndentations(`
+	{{ .Prompt }}
+	` + "```" + `
+	{{ .Code }}
+	` + "```" + `
+`)))
+
+// GenerateTestsForFile implements Model.
```

Comment:
Always copy the docstring of the interface and keep it in sync. we do this internally too.